magenta-lib/src/main/scala/magenta/graph/Graph.scala

```diff
@@ -113,13 +180,17 @@ case class Graph[T](edges: Set[Edge[T]]) {
   lazy val toList: List[T] = nodeList.flatMap(_.maybeValue)
 
   override def toString: String = {
```

Comment:
Possibly. This was a big improvement over the existing case class output without the complexity of having to work out how to render a graph with ascii art.

It's useful to have the priority here - there is no reason why you can't have the same priority twice which could result in `toString` output that looks the same but two case classes that fail equality.